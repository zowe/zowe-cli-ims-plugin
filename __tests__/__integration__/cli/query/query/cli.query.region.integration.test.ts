/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

// Test environment will be populated in the "beforeAll"
import { ITestEnvironment } from "../../../../__src__/environment/doc/response/ITestEnvironment";
import { TestEnvironment } from "../../../../__src__/environment/TestEnvironment";
import { runCliScript } from "../../../../__src__/TestUtils";

let TEST_ENVIRONMENT: ITestEnvironment;
describe("ims query region", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "query_region_command",
            installPlugin: true,
            skipProperties: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should display the query region help", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/query_region_help.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
