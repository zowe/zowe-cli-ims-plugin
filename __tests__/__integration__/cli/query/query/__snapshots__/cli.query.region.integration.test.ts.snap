// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ims query region should display the query region help 1`] = `
"===============IMS QUERY REGION HELP===============

 COMMAND NAME
 ------------

   region | reg

 DESCRIPTION
 -----------

   Query an IMS region.

 USAGE
 -----

   zowe ims query region [options]

 OPTIONS
 -------

   --dc  (boolean)

      Displays only the DC subset of the output

      Default value: true

   --region  (boolean)

      Displays only the REGION subset of the output. The display consists of active
      regions

      Default value: true

   --route  | --rt (array)

      Specifies the routes to return.

 IMS CONNECTION OPTIONS
 ----------------------

   --host  | -H (string)

      The IMS Operations API server host name.

   --port  | -P (number)

      The IMS Operations API server port.

   --ims-connect-host  | --ich (string)

      The hostname of your instance of IMS Connect. This is typically the hostname of
      the mainframe LPAR where IMS Connect is running.

   --ims-connect-port  | --icp (number)

      The port of your instance of IMS Connect. This port can be found in your IMS
      Connect configuration file on the mainframe.

   --plex  | -x (string)

      The name of the IMS plex.

   --user  | -u (string)

      The web server user name where the IMS Operations API resides.

   --password  | --pass (string)

      The web server user password where the IMS Operations API resides.

   --base-path  | --bp (string)

      The base path for your API mediation layer instance. Specify this option to
      prepend the base path to all z/OSMF resources when making REST requests. Do not
      specify this option if you are not using an API mediation layer.

 PROFILE OPTIONS
 ---------------

   --ims-profile  | --ims-p (string)

      The name of a (ims) profile to load for this command execution.

   --base-profile  | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --token-type  | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value  | --tv (string)

      The value of the token to pass to the API.

 RESPONSE FORMAT OPTIONS
 -----------------------

   --response-format-filter  | --rff (array)

      Filter (include) fields in the response. Accepts an array of field/property
      names to include in the output response. You can filter JSON objects properties
      OR table columns/fields. In addition, you can use this option in conjunction
      with '--response-format-type' to reduce the output of a command to a single
      field/property or a list of a single field/property.

   --response-format-type  | --rft (string)

      The command response output format type. Must be one of the following:

      table: Formats output data as a table. Use this option when the output data is
      an array of homogeneous JSON objects. Each property of the object will become a
      column in the table.

      list: Formats output data as a list of strings. Can be used on any data type
      (JSON objects/arrays) are stringified and a new line is added after each entry
      in an array.

      object: Formats output data as a list of prettified objects (or single object).
      Can be used in place of \\"table\\" to change from tabular output to a list of
      prettified objects.

      string: Formats output data as a string. JSON objects/arrays are stringified.

      Allowed values: table, list, object, string

   --response-format-header  | --rfh (boolean)

      If \\"--response-format-type table\\" is specified, include the column headers in
      the output.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a the group

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Query information for regions on route IMS1:

      $ zowe ims query region \\"IMS1\\"

   - Query information for regions on routes IMS1 and IMS2:

      $ zowe ims query region \\"IMS1 IMS2\\"

   - Query DC and region information for regions on routes IMS1
   and IMS2:

      $ zowe ims query region \\"IMS1 IMS2\\" --dc true --region true

   - Query information for regions specifying optional
   connection parameters:

      $ zowe ims query region --user \\"username\\" --pass \\"pass1234\\" --host \\"localhost\\" --port 8080 --ich \\"zos1\\" --icp 9999 --plex \\"PLEX1\\"

"
`;
